cpp/thirdparty/flatbuffers/README.md

```diff
@@ -40,3 +41,126 @@ index 955738067..fccce42f6 100644
    #endif // __has_include
  #endif // !FLATBUFFERS_HAS_STRING_VIEW
 ```
+
+```diff
+diff --git a/cpp/thirdparty/flatbuffers/include/flatbuffers/base.h b/cpp/thirdparty/flatbuffers/include/flatbuffers/base.h
+index fccce42f6..316eb2259 100644
+--- a/cpp/thirdparty/flatbuffers/include/flatbuffers/base.h
++++ b/cpp/thirdparty/flatbuffers/include/flatbuffers/base.h
+@@ -1,6 +1,11 @@
+ #ifndef FLATBUFFERS_BASE_H_
+ #define FLATBUFFERS_BASE_H_
+ 
++// Move this vendored copy of flatbuffers to a private namespace,
++// but continue to access it through the "flatbuffers" alias.
++namespace arrow_thirdparty_flatbuffers {}
```

Comment:
We already have a `arrow_vendored` namespace, so ideally this would be `arrow_vendored::flatbuffers`

Reply:
Hmm, actually that is not right, because it would then be exposed through the linker script:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/symbols.map

Perhaps `arrow_vendored_private::flatbuffers`?

Reply:
Thanks for your response, and that seems reasonable to me. I will make that change shortly.

Reply:
OK, changes made. Please take another look.